public/src/component/App.js

```diff
@@ -0,0 +1,76 @@
+import { html, render, useEffect, useState } from 'https://unpkg.com/htm@3.1.0/preact/standalone.module.js'
+import { Event } from './Event.js'
+import { Header } from './Header.js'
+import { Filter, filterFunc } from './Filter.js'
+
+function App() {
+    const [event, setEvent] = useState(null);
+    const [events, setEvents] = useState([]);
+    const [isStreaming, setStreaming] = useState(true);
+    const [automaticScrollEnabled, setAutomaticScrollEnabled] = useState(false);
+
+    const [textFilter, setTextFilter] = useState(null);
+    const [operationFilter, setOperationFilter] = useState(null);
+    const [limit, setLimit] = useState(100);
+
+    useEffect(() => {
+        const eventSource = new EventSource('/sse/event');
+
+        eventSource.addEventListener('error', () => setStreaming(false));
+        eventSource.addEventListener('event', (e) => setEvent(JSON.parse(e.data)));
+    }, []);
+
+    useEffect(() => {
+        // Keep maximum 500 events in memory.
+        const filteredEvents = events.filter(event => event !== null).slice(-500);
+        setEvents([...filteredEvents, event]);
+
+        if (automaticScrollEnabled) {
+            window.scrollTo(0, document.body.scrollHeight);
+        }
+    }, [event]);
+
+    return html`
+        <${Header} />
```

Comment:
Maybe you can add some information about the database and the name of the collection currently watched.